froide/foirequest/views/message.py

```diff
@@ -151,6 +157,311 @@ def add_postal_message(request, slug):
     )
 
 
+@allow_write_foirequest
+def upload_postal_message_create(request, foirequest):
+    message = FoiMessage(request=foirequest, kind=MessageKind.POST)
+    # hack, prevent: null value in column "timestamp" of relation "foirequest_foimessage" violates not-null constraint
+    # this will be set to "proper" (noon) when finally submitted
+    message.timestamp = datetime.datetime.now()
```

Comment:
oof, hard to tell (for me). i think it *shouldn't*, because, as stated in the comment, it's a temporary non-null replacement, that shouldn't exist for too long.
but speaking of timezone problems, `registered_mail_date` from your other comment might cause problems? it's supposed to be a "timeless" date, so it gets set to midnight. i found `date` field logic hard to understand...

Reply:
@krmax44 suggest adding a constraint that allows this field to be nullable as long is_draft UNLESS this is the only timestamp/created_at-ish info